### PR TITLE
feat: add license setting get/put api

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-10252

Add `/license/setting` GET/PUT HTTP API for setting the alarm watermark of license.

image

### <samp>🤖 Generated by Copilot at f03d4d0</samp>

This pull request adds a new HTTP API endpoint for getting and updating the license setting of EMQ X. It also refactors the license schema and code to use the hocon schema format and macros, and updates the HTTP API documentation accordingly. The files `emqx_license_http_api.erl`, `emqx_license_schema.erl`, `emqx_license.erl`, `emqx_license_schema.hocon`, and `emqx_license_http_api.hocon` are modified.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: lib-ee/emqx_license/src/emqx_license_http_api.erl [hunk 1]
```diff
@@ -50,19 +54,18 @@ schema("/license") ->
                 )
             }
         },
+        %% FIXME: It's a update action, should use put instead of post in 5.2.0
```
Comment (zmstone) [inline lib-ee/emqx_license/src/emqx_license_http_api.erl:57]:
```suggestion
        %% TODO(5.x): It's a update action, should use PUT instead
```

File: lib-ee/emqx_license/src/emqx_license_http_api.erl [hunk 2]
```diff
@@ -117,3 +142,40 @@ error_msg(Code, Msg) ->
...
+        {ok, _} ->
+            ?SLOG(info, #{msg => "updated_license_setting"}),
+            '/license/setting'(get, undefined)
+    end.
+
+fields(key_license) ->
+    Key = lists:keyfind(key, 1, emqx_license_schema:fields(key_license)),
+    [
+        Key,
+        %% FIXME: remove when 5.2.0
```
Comment (zmstone) [inline lib-ee/emqx_license/src/emqx_license_http_api.erl:165]:
```suggestion
        %% TODO(5.x): remove when we decide to break compatibility
```

File: lib-ee/emqx_license/src/emqx_license_http_api.erl [hunk 3]
```diff
@@ -117,3 +142,40 @@ error_msg(Code, Msg) ->
...
+            example => <<"75%">>,
+            deprecated => true,
+            desc => ?DESC(connection_low_watermark_field_deprecated)
+        }},
+        {connection_high_watermark, #{
+            type => emqx_schema:percent(),
+            example => <<"80%">>,
+            deprecated => true,
+            desc => ?DESC(connection_high_watermark_field_deprecated)
+        }}
```
Comment (zmstone) [inline lib-ee/emqx_license/src/emqx_license_http_api.erl:177]: no need to keep them, there should be no compatibility issue.

Comment (zhongwencool) [reply lib-ee/emqx_license/src/emqx_license_http_api.erl]: DONE

**Review by zmstone — APPROVED** [review]

**Pull request merged**